bioeng-example-lesson.md

```diff
@@ -0,0 +1,145 @@
+title: Controlling a quadcopter with your mind
+subtitle: Analyzing brain data
+minutes: 30
+---
+
+> ## Learning Objectives {.objectives}
+> * Apply the core software carpentry skills in the field of biomedical engineering.
+> * Loading .mat-files 
+> * Logical indexing
+
+Some researchers were able to control a quadrocoptor with [nothing but thoughts](https://www.youtube.com/watch?v=-h3kiws4I54). 
+Now we want to replicate this work.
+<img src="img/quadrocoptor.png" style="height:350px">
+
+To do this, first we need to measure brain activity. 
+This can be done using EEG (electroencephalogram). A test participant wears a cap with 
+embedded electrodes. The participant then repeatedly moves their left or right hand or no hand at all, 
+while we measure the electric field with the electrodes. Luckily this data has 
+already been recorded and can be downloaded [here](https://github.com/EwanNurse/A-Generalizable-BCI-using-Machine-Learning-for-Feature-Discovery/tree/master/data). We'll be using 'EEG_ParticipantB_30_09_2013.mat'.
+
+Once we have downloaded the first data file, we want to:
+
+* load this data into memory
+* classify the data into right hand movement, left hand movement, no movement
+* calculate the average brain activity on each electrode for each class
+* plot the result using some code we found on the Internet 
+* find out which electrodes are most active
+
+First, let's have a look at the data file we downloaded. 
+The file is a .mat-file. This is a special MATLAB file type that can be used
+to store multiple variables at once. 
+To load the stored variables back into memory, type 
+
+~~~ {.matlab}
+load('data/EEG_ParticipantB_28_09_2013.mat');
+~~~
+
+If we have a look at our Workspace, we see that two new variables have been loaded:
+data_epoch_B and data_key_B.
+
+The recorded data is stored in `data_epoch_B`, the classification key is
+stored in `data_key_B`. 
+
+Each column in data_epoch_B contains the recorded data of one trial, 
+corresponding to one value in data_key_B (1 = left hand press, 2 = right 
+hand press, 3 = no hand press).
+The first 100 values in each column are the recordings during 0.4 ms 
+of one electrode, the second 100 values of the second electrode and 
+so on. 
+<img src="img/brain_datafiles.png" style="height:350px">
+
+We now want to have not one, but three arrays, one for each condition 
+(1 = left hand press, 2 = right hand press, 3 = no hand press).
+MATLAB provides us with a handy way of obtaining all indexes for fields that 
+meet a certain condition. This method is called logical indexing.
+
+For example, we have recorded how many mice our cat caught each day for one week.
+
+~~~ {.matlab}
+mice = [4 5 2 6 7 8 3];
+~~~
+
+
+Now if we want to know how often our cat has brought home 4 or more mice, we 
+can check that by typing:
+
+~~~ {.matlab}
+mice>=4
+~~~
+
+The response is a vector with ones wherever this condition is true.
+
+~~~ {.output}
+ans = 1     1     0     1     1     1     0
+~~~
+
```

Comment:
Maybe there needs to be a little more information about logical vectors, and how they work (a callout box with a few more examples, perhaps?)